### ISSUE TITLE
Using INDEX on a particles shader, breaks the shader

### ISSUE THREAD

**Issue by eon-s** [godotengine/godot#8731]
**Godot version: Master ed6baffc7**

**Issue description:**
Just using INDEX on a particles shader, breaks the shader.

INDEX is mentioned here https://github.com/godotengine/godot/blob/4286aef69313e048fa91710c35456c08a252fd3c/servers/visual/shader_types.cpp#L205 , I suppose refers to the same index as the Particles node .

The terminal shows some code, then:
```
ERROR: _display_error_with_code: ParticlesShaderGLES3: Vertex Program Compilation Failed:
0(173) : error C1008: undefined variable "index"

   At: drivers/gles3/shader_gles3.cpp:167.
ERROR: get_current_version: Method/Function Failed, returning: __null
   At: drivers/gles3/shader_gles3.cpp:361.
ERROR: bind: Condition ' !version ' is true. returned: false
   At: drivers/gles3/shader_gles3.cpp:122.
ERROR: _get_uniform: Condition ' !version ' is true. returned: -1
   At: drivers/gles3/shader_gles3.h:376.

```

_Trying to create a variable named INDEX in the shader code gives an error of redefinition of variable._

Then spams this error:

```
mesege
ERROR: _gl_debug_print: GL ERROR: Source: OpenGL	Type: Error	ID: 1282Severity: High	Message: GL_INVALID_OPERATION error generated. No active program.
   At: drivers/gles3/rasterizer_gles3.cpp:123.

```

**Steps to reproduce:**
Use this simple shader code on a Particles node (with assigned mesh):

```
shader_type particles;

void vertex(){
	CUSTOM.y += 10.0*DELTA;
	int a = INDEX;
	
	TRANSFORM = mat4(vec4(1,0,0,0),vec4(0,1,0,0),vec4(0,1,0,0),vec4(CUSTOM.xyz,1));
}
```

**Comment by ippan** [comment]
maybe try to use NUMBER in particles shader

**Comment by eon-s** [comment]
@ippan NUMBER may be the amount of particles in the emitter (is the same for all particles), not the particle index, this seems to be a variable problem in some shader functions.

**Comment by ippan** [comment]
@eon-s yes, you are right, it's not the index ^^

**Comment by eon-s** [comment]
@ippan Your fix mentioned here https://github.com/godotengine/godot/issues/8751#issuecomment-302736472
seems to work :smiley: 

![indexfix](https://cloud.githubusercontent.com/assets/10215987/26255871/61e66da2-3c91-11e7-8b3a-66a19be2d51e.gif)

**Comment by vnen** [comment]
Fixed by #8835.
